php-checks/src/main/java/org/sonar/php/checks/ConstructWithParenthesesCheck.java

```diff
@@ -0,0 +1,132 @@
+  }
+
+  @Override
+  public void visitBreakStatement(BreakStatementTree tree) {
+    raiseIssueOnParenthesizedExpression(tree.argument(), tree, "break");
+    super.visitBreakStatement(tree);
+  }
+
+  @Override
+  public void visitReturnStatement(ReturnStatementTree tree) {
+    raiseIssueOnParenthesizedExpression(tree.expression(), tree, "return");
+    super.visitReturnStatement(tree);
+  }
+
+  @Override
+  public void visitThrowStatement(ThrowStatementTree tree) {
+    raiseIssueOnParenthesizedExpression(tree.expression(), tree, "throw");
+    super.visitThrowStatement(tree);
+  }
+
+  @Override
+  public void visitYieldExpression(YieldExpressionTree tree) {
+    String constructName = tree.fromToken() != null ? "yield from" : "yield";
+    ExpressionTree elementToCheck = tree.key() != null ? tree.key() : tree.value();
+    raiseIssueOnParenthesizedExpression(elementToCheck, tree, constructName);
+    super.visitYieldExpression(tree);
+  }
+
+  @Override
+  public void visitContinueStatement(ContinueStatementTree tree) {
+    raiseIssueOnParenthesizedExpression(tree.argument(), tree, "continue");
+    super.visitContinueStatement(tree);
+  }
+
+  private void raiseIssueOnParenthesizedExpression(@Nullable ExpressionTree expression, Tree raiseIssueOn, String constructName) {
+    if (isParenthesizedExpression(expression)) {
+      newIssue(raiseIssueOn, String.format(MESSAGE, constructName));
+    }
+  }
+
+  private void raiseIssueOnParenthesizedExpression(@Nullable ExpressionTree expression, Tree raiseIssueOnStart, Tree raiseIssueOnEnd, String constructName) {
+    if (isParenthesizedExpression(expression)) {
+      context().newIssue(this, raiseIssueOnStart, raiseIssueOnEnd, String.format(MESSAGE, constructName));
+    }
+  }
+
+  private static boolean isParenthesizedExpression(@Nullable ExpressionTree expression) {
+    return expression != null && expression.is(Tree.Kind.PARENTHESISED_EXPRESSION);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    String constructName = getConstructName(tree.callee());
+    if (isConstructFunction(constructName) && isParenthesized(tree)) {
+      newIssue(tree, String.format(MESSAGE, constructName));
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  private static boolean isParenthesized(FunctionCallTree tree) {
+    return tree.callArguments().size() == 1 && isParenthesizedArgument(tree.callArguments().get(0).value());
+  }
+
+  private static boolean isParenthesizedArgument(ExpressionTree expression) {
+    return expression.is(Tree.Kind.PARENTHESISED_EXPRESSION)
+      || (expression instanceof BinaryExpressionTree && isParenthesizedArgument(((BinaryExpressionTree) expression).leftOperand()));
+  }
+
+  private static boolean isConstructFunction(@Nullable String constructName) {
+    return constructName != null && CONSTRUCT_FUNCTION.contains(constructName);
+  }
+
+  @CheckForNull
+  private static String getConstructName(ExpressionTree callee) {
```

Comment:
In fact, this method is not returning the construct name, it returns a callee name.